include/d/actor/d_a_npc_bs1.h

```diff
@@ -1,101 +1,192 @@
-    void getHead_y() {}
+    int getHeadJntNum() { return m_head_jnt_num; }
+    int getHead_x() { return mJntCtrl.getHead_x(); }
+    int getHead_y() { return mJntCtrl.getHead_y(); }
     void getPayRupee() {}
     void incAttnSetCount() {}
     void setAction(int (daNpc_Bs1_c::*)(void*), void*) {}
-    void setAttentionBasePos(cXyz) {}
+    void setAttentionBasePos(cXyz value) { m718 = value;}
     void setBuyItem(unsigned char) {}
     void setBuyItemMax(unsigned char) {}
-    void setEyePos(cXyz) {}
+    void setEyePos(cXyz other) { eyePos = other; }
     void setPayRupee(short) {}
 
-    void XyEventCB(int);
-    void initTexPatternAnm(bool);
+    s16 XyEventCB(int);
+    BOOL initTexPatternAnm(bool);
     void playTexPatternAnm();
     void setAnm(signed char);
     void setTexAnm(signed char);
     void setAnmFromMsgTag();
-    void chkAttention(cXyz, short);
+    BOOL chkAttention(cXyz, short);
     void eventOrder();
     void checkOrder();
-    void next_msgStatus(unsigned long*);
+    u16 next_msgStatus(unsigned long*);
     void getMsg();
     void setCollision();
     void talkInit();
-    void shopMsgCheck(unsigned long);
+    u32 shopMsgCheck(unsigned long);
     void getDefaultMsg();
     void shopStickMoveMsgCheck(unsigned long);
-    void checkBeastItemSellMsg(unsigned long);
+    BOOL checkBeastItemSellMsg(u32);
     void normal_talk();
     void shop_talk();
-    void talk();
+    u16 talk();
     void createShopList();
-    void isSellBomb();
-    void CreateInit();
+    BOOL isSellBomb();
+    BOOL CreateInit();
     void setAttention(bool);
     void lookBack();
-    void wait01();
-    void talk01();
-    void wait_action(void*);
-    void getdemo_action(void*);
-    void evn_talk_init(int);
-    void evn_continue_talk_init(int);
-    void evn_talk();
-    void evn_jnt_lock_init(int);
-    void evn_wait_init(int);
-    void evn_wait();
-    void evn_set_anm_init(int);
-    void evn_praise_init();
-    void evn_mantan_init();
-    void privateCut();
-    void event_action(void*);
+    BOOL wait01();
+    BOOL talk01();
+    BOOL wait_action(void*);
+    BOOL getdemo_action(void*);
+    BOOL evn_talk_init(int);
+    BOOL evn_continue_talk_init(int);
+    BOOL evn_talk();
+    BOOL evn_jnt_lock_init(int);
+    BOOL evn_wait_init(int);
+    BOOL evn_wait();
+    BOOL evn_set_anm_init(int);
+    BOOL evn_praise_init();
+    BOOL evn_mantan_init();
+    BOOL privateCut();
+    BOOL event_action(void*);
     BOOL _draw();
     BOOL _execute();
     BOOL _delete();
     s32 _create();
-    void CreateHeap();
+    BOOL CreateHeap();
+
+    typedef BOOL (daNpc_Bs1_c::*ActionFunc)(void*);
 
 public:
-    /* 0x290 */ u8 m290[0x748 - 0x290];
+    /* 0x290 */ request_of_phase_process_class mphs;
+    /* 0x298 */ mDoExt_McaMorf* mpMorf;
+    /* 0x29C */ u32 m29C;
+    /* 0x2A0 */ J3DModel* mpModelArr[3];
+    /* 0x2AC */ J3DAnmTexPattern* m_head_tex_pattern;
+    /* 0x2B0 */ mDoExt_btpAnm mBtpAnm;
+    /* 0x2C4 */ u8 i_frame;
```

Comment:
where did the name `i_frame` come from? that looks like the style they use for function parameters, I've never seen it used for member variables

Reply:
I probably just copied the parameter name and put it there. I'll swap it to `mFrame` during the rest of the PR review changes.